kong/clustering/services/sync/rpc.lua

```diff
@@ -400,17 +433,27 @@ function sync_once_impl(premature, retry_count)
   local current_version = get_current_version()
   if current_version >= latest_notified_version then
     ngx_log(ngx_DEBUG, "version already updated")
-    return
+    return sync_handler() -- call get_delta once more to report to the CP that we are up-to-date
   end
 
   -- retry if the version is not updated
   retry_count = retry_count or 0
+  -- we do not count a timed out sync. just retry
+  if err ~= "timeout" then
+    retry_count = retry_count + 1
+  end
```

Comment:
Please refer to the tests in https://github.com/Kong/kong/pull/14174 to ensure it will not break the cases.